engine/api/project.go

```diff
@@ -83,74 +163,22 @@ func (api *API) getProjectsHandler() service.Handler {
 			return err
 		}
 
+		var groupIDs []int64
+		if requestedUser == nil {
+			groupIDs = getAPIConsumer(ctx).GetGroupIDs()
+		} else {
+			groupIDs = requestedUser.GetGroupIDs()
+		}
+
 		pKeys := projects.Keys()
-		perms, err := permission.LoadProjectMaxLevelPermission(ctx, api.mustDB(), pKeys, getAPIConsumer(ctx).GetGroupIDs())
+		perms, err := permission.LoadProjectMaxLevelPermission(ctx, api.mustDB(), pKeys, groupIDs)
```

Comment:
Looks like this code is useful only if withPermission, also maintainer an admin can have no permission on project based on groups but we should add read permission for maintainer and all permissions for admin.

Reply:
We always need permission. BWT i fixed it for admin, and maintainer permissions